MushroomObserver/mushroom-observer
# iNat Import

PR aims to implement the minimum acceptable functionality for NEMF for importing iNat observations
- Allows an MO user to import a list of that User's iNat observation(s) to MO via the iNatImport form.
- Implements the [Basic Inat Import milestone](https://github.com/MushroomObserver/mushroom-observer/milestone/3)

### Note
- It uses an inefficient way of giving users feedback about jobs. I didn't know how to do the js-Turbo 🪄 voodoo. I think that's enough for NEMF because I removed the UI for importing all a user's observations.
- It doesn't rate limit. I think that's OK for NEMF for the the same reason.

### Questions for reviewers
- Is more functionality needed?
- Appearance
- General organization
- Better names for classes, methods, variables, ex: ImportedInatObs
- Any ideas about shortening/dividing long classes/methods
- Any ideas about fixing ABC violations
- Any other criticism is welcome.

### Manual Test Protocol
- Create a sacrificial iNat observation, making sure it has lat/lng. I suggest also selecting Obscured Geoprivacy.
- In MO: Make sure SolidQueue is running, Create Observation, Import from iNat http://localhost:3000/observations/inat_imports/new
- Fill in the fields, using the iNat observation # of the sacrificial iNat obs, and Submit
Result: redirects to Observation Index (to be change to Inat Import Job Tracker)
- Wait 10 seconds, then refresh screen
Expected result: Index includes a new Observation
- Goto the new Observation. and review it. The Location should make sense. If the iNat obs had Obscured Geoprivacy, then MO coordinates should be hidden from the public.
- Goto/refresh the sacrificial iNat Obs
Expected result: 
Description includes imported to MO and date of the import; 
Observation Fields include Mushroom Observer URL linking to the newly created **Local** observation. 
(In the production environment, it links to the live db.)
- Delete the sacrificial iNat Obs
